tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -452,6 +452,26 @@ public void CustomSamplingRules(string value, string expected)
             settings.CustomSamplingRules.Should().Be(expected);
         }
 
+        [Theory]
+        [InlineData("glob", SamplingRulesFormat.Glob)]     // exact match
+        [InlineData("Glob", SamplingRulesFormat.Glob)]     // case-insensitive
+        [InlineData(" glob ", SamplingRulesFormat.Glob)]   // trim whitespace
+        [InlineData("regex", SamplingRulesFormat.Regex)]   // exact match
+        [InlineData("RegEx", SamplingRulesFormat.Regex)]   // case-insensitive
+        [InlineData(" regex ", SamplingRulesFormat.Regex)] // trim whitespace
+        [InlineData("none", SamplingRulesFormat.Unknown)]  // invalid
+        [InlineData("1", SamplingRulesFormat.Unknown)]     // invalid
+        [InlineData(null, SamplingRulesFormat.Regex)]      // null or empty or whitespace defaults to regex
+        [InlineData("", SamplingRulesFormat.Regex)]        // null or empty or whitespace defaults to regex
+        [InlineData("  ", SamplingRulesFormat.Regex)]      // null or empty or whitespace defaults to regex
+        public void CustomSamplingRulesFormat(string value, string expected)
+        {
+            var source = CreateConfigurationSource((ConfigurationKeys.CustomSamplingRulesFormat, value));
+            var settings = new TracerSettings(source);
```

Comment:
A nit, but if you wanted to be sure about the telemetry, you could do something like this:

```suggestion
            var telemetry = new ConfigurationTelemetry();
            var settings = new TracerSettings(source, telemetry);
```

And then you could assert on the contents of `telemetry` to make sure it's recorded as you expect 🙂

Reply:
Thanks! I _think_ I figured out how to do this, but let me know if there's a better way.